File: ima-backend/src/main/java/com/techsolution/ima_backend/services/CompanyService.java
package com.techsolution.ima_backend.services;

import com.techsolution.ima_backend.dtos.request.CompanyRequest;
import com.techsolution.ima_backend.dtos.response.CompanyResponse;


import java.util.List;

public interface CompanyService {

    CompanyResponse createCompany(Long userId, CompanyRequest companyRequest);
    List<CompanyResponse> getAllCompanies();
    CompanyResponse getCompanyById(Long companyId);
    CompanyResponse updateCompany(Long companyId, CompanyRequest companyRequest);
    void deleteCompany(Long companyId);
}

File: ima-frontend-react/src/services/CompanyService.js
import securedAxiosInstance from "./SecuredAxiosInstance";

const REST_API_COMPANIES = '/companies';

export const listCompanies = () => securedAxiosInstance.get(`${REST_API_COMPANIES}`);

export const getCompany = (companyId) => securedAxiosInstance.get(`${REST_API_COMPANIES}/${companyId}`);

export const deleteCompany = (companyId) => securedAxiosInstance.delete(`${REST_API_COMPANIES}/${companyId}`);

export const updateCompany = (companyId, companyRequest) => securedAxiosInstance.put(`${REST_API_COMPANIES}/${companyId}`, companyRequest);

export const createCompany = (userId, companyRequest) => 
    securedAxiosInstance.post(
        REST_API_COMPANIES,
        companyRequest,
        {
            params: {userId}
        }
    );